config/config.go

```diff
@@ -57,25 +69,63 @@ func FindConfig(fs afero.Fs, configFile string) ([]byte, int, error) {
 	return b, v, nil
 }
 
+func findConfig(fs afero.Fs, configFile string) ([]byte, int, error) {
+	f, err := fs.Open(configFile)
+	if err != nil {
+		return nil, 0, errs.WrapUser(err, "unable to open config file")
+	}
+
+	b, e := ioutil.ReadAll(f)
+	if e != nil {
+		return nil, 0, errs.WrapUser(e, "unable to read config")
+	}
+
+	v, err := detectVersion(b)
+	if err != nil {
+		return nil, 0, err
+	}
+	logrus.Debugf("config file version: %#v\n", v)
+	return b, v, nil
+}
+
 func FindAndReadConfig(fs afero.Fs, configFile string) (*v2.Config, error) {
 	b, v, err := FindConfig(fs, configFile)
 	if err != nil {
 		return nil, err
 	}
+
 	switch v {
-	case 1:
-		c, err := v1.ReadConfig(b)
-		if err != nil {
-			return nil, err
+	case 1: //Upgrade the config version
+		if strings.Compare(configFile, "fogg.json") == 0 {
+			c, err := v1.ReadJsonConfig(b)
+			if err != nil {
```

Comment:
```
			if err != nil {
				return nil, err
			}
			c2, err := UpgradeConfigVersion(c)
			return c2, err
```
is duplicated for both branches of this conditional, is there a way to avoid this duplication?